lib/ansible/plugins/strategy/__init__.py

```diff
@@ -656,7 +656,9 @@ def _process_pending_results(self, iterator, one_pass=False, max_passes=None):
                         # otherwise depending on the setting either error or warn
                         host_state = iterator.get_state_for_host(original_host.name)
                         for notification in result_item['_ansible_notify']:
+                            found_handler = False
                             for handler in self.search_handlers_by_notification(notification, iterator):
+                                found_handler = True
```

Comment:
I would probably just pre-set the `handler` variable to some dummy value (e.g. `ansible.utils.Sentinel`) before the loop and then after the loop check for that value to indicate no handlers were found.

Reply:
Sure, thanks! I've just pushed a commit with this change.